luten-scheduling/src/instances/random.rs

```diff
@@ -0,0 +1,202 @@
+use rand::{Rng, thread_rng};
+use rand::distributions::{IndependentSample, Sample};
+use rand::distributions::normal::Normal;
+use rand::distributions::range::Range;
+use std::collections::HashMap;
+
+use types::*;
+use WorkDay::*;
+use util::clamp;
+
+pub struct RatingDistribution<T, U, V> where
+    T: IndependentSample<f64>,
+    U: IndependentSample<f64>,
+    V: IndependentSample<f64>,
+{
+    pub available_blocks_per_day: u16,
+    // distribution used to sample overall number of slots rated as `Good` or `Tolerable`
+    pub rated_slots: T,
+    // proportion of slots rated good vs tolerable
+    pub good_slot_percentage: U,
+    // percentage of slots rated together as a block of four slots
+    pub block_percentage: V,
+}
+
+impl RatingDistribution<Normal, Range<f64>, Range<f64>> {
+    pub fn default() -> Self {
+        use rand::distributions::normal::Normal;
+        use rand::distributions::range::Range;
+        Self {
+            available_blocks_per_day: 4,
+            rated_slots: Normal::new(10.0, 2.5),
+            good_slot_percentage: Range::new(0.6, 0.9),
+            block_percentage: Range::new(0.5, 0.7),
+        }
+    }
+}
+
+impl<T, U, V> Sample<SlotAssignment> for RatingDistribution<T, U, V> where
+    T: IndependentSample<f64>,
+    U: IndependentSample<f64>,
+    V: IndependentSample<f64>,
+{
+    fn sample<R: Rng>(&mut self, _: &mut R) -> SlotAssignment {
+        unimplemented!()
+    }
+}
+
+impl<T, U, V> IndependentSample<SlotAssignment> for RatingDistribution<T, U, V> where
+    T: IndependentSample<f64>,
+    U: IndependentSample<f64>,
+    V: IndependentSample<f64>,
+{
+    fn ind_sample<R: Rng>(&self, rng: &mut R) -> SlotAssignment {
+
+        /// Generates a `ŚlotAssignment` where x = 4 * `good_blocks` + `good_slots`
+        /// slots have been marked as `SlotRating::Good`
+        /// and y = 4 * `tolerable_blocks` + `tolerable_slots`
+        /// slots have been marked as `SlotRating::Tolerable`
+        /// and where z = `good_blocks` + `tolerable_blocks`
+        /// blocks of 4 coherent slots appear
```

Comment:
I'm a bit confused by the formatting of this comment... You can/should use markdown lists if you have list like thingies.

```
/// Generates:
/// 
/// - foo
/// - bar
```